File: spec/cpp_stl_98/test_debug_array_user.cpp
#include <boost/test/unit_test.hpp>
#include "debug_array_user.h"
#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_debug_array_user) {
    std::ifstream ifs("src/fixed_struct.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    debug_array_user_t* r = new debug_array_user_t(&ks);

    // --debug implies --no-auto-read
    r->_read();

    BOOST_CHECK_EQUAL(r->one_cat()->meow(), 80);
    BOOST_CHECK_EQUAL(r->array_of_cats()->size(), 3);
    BOOST_CHECK_EQUAL(r->array_of_cats()->at(0)->meow(), 65);
    BOOST_CHECK_EQUAL(r->array_of_cats()->at(1)->meow(), 67);
    BOOST_CHECK_EQUAL(r->array_of_cats()->at(2)->meow(), 75);

    delete r;
}

File: spec/rust/tests/test_debug_array_user.rs
use std::fs;
extern crate kaitai;
use self::kaitai::*;
use rust::formats::debug_array_user::*;

#[test]
fn test_debug_array_user() -> KResult<()> {
    let bytes = fs::read("../../src/fixed_struct.bin").unwrap();
    let _io = BytesReader::from(bytes);
    let r: OptRc<DebugArrayUser> = DebugArrayUser::read_into(&_io, None, None)?;

    assert_eq!(*r.one_cat().meow(), 0x50);
    assert_eq!(*r.array_of_cats()[0].meow(), 0x41);
    assert_eq!(*r.array_of_cats()[1].meow(), 0x43);
    assert_eq!(*r.array_of_cats()[2].meow(), 0x4b);
    Ok(())
}
